src/Functions/FunctionBase64Conversion.h

```diff
@@ -85,6 +85,85 @@ struct TryBase64Decode
     }
 };
 
+struct Base64UrlEncode : Base64Encode
+{
+    static constexpr auto name = "base64UrlEncode";
+
+    static size_t perform(const std::span<const UInt8> src, UInt8 * dst)
+    {
+        auto out_len = Base64Encode::perform(src, dst);
+
+        // Do postprocessing as described in https://datatracker.ietf.org/doc/html/rfc4648#page-7
+        for (size_t i = 0; i < out_len; ++i)
+        {
+            switch (dst[i])
+            {
+            case '/':
+                dst[i] = '_';
+                break;
+            case '+':
+                dst[i] = '-';
+                break;
+            case '=': // stop when padding is detected
+                return i;
+            default:
+                break;
+            }
+        }
+        return out_len;
+    }
+};
+
+struct Base64UrlDecode : Base64Decode
+{
+    static constexpr auto name = "base64UrlDecode";
+
+    static size_t perform(const std::span<const UInt8> src, UInt8 * dst)
+    {
+        std::vector<UInt8> tmp{};
+        // insert padding to please alcomp library
+        auto size = src.size();
+        auto remainder = size % 4;
+        switch (remainder)
+        {
+            case 0:
+                break; // no padding needed
+            case 1:
+                break; // invalid input, let it be detected by alcomp library
```

Comment:
Not sure if I understand: why is that invalid input exactly? Would we achieve the same result (probably an exception or error returned by aklomp) if we simply pad the input like in cases `2` and `3`?

Also: `alcomp` --> `aklomp`

Reply:
Yes, we can do it to behave similarly in all cases. So, remainder one length (that corresponds to triple-"=" padding) is impossible to be generated by base64 encoding algorithm design, and thus by aklomp library.
If the user submits a string for url64 decoding, which length % 4 == 1, it is certainly an invalid input, and this is detected by aklomp library itself. I can write a comment on this.